docs/dev/run_the_etl.rst

```diff
@@ -11,24 +11,182 @@ just want to use already processed data.
 
 These instructions assume you have already gone through the :ref:`dev_setup`.
 
-There are two main scripts involved in the PUDL processing pipeline:
-
-1. ``ferc_to_sqlite`` :doc:`converts the FERC Form 1 DBF/XBRL files <clone_ferc1>` into
-   a single large `SQLite <https://sqlite.org>`__ database so that the data is easier to
-   extract, and so all of the raw FERC Form 1 data is available in a modern format. You
-   must run this script before you can run ``pudl_etl``.
-2. ``pudl_etl`` coordinates the "Extract, Transform, Load" process that processes
+Dagster
+-------
+PUDL uses `Dagster <https://dagster.io/>`__ to orchestrate data pipelines. Dagster
+makes it easy to manage data dependences, parallelize processes, cache results
+and handle IO. If you are planning on contributing to PUDL, it is recommended you
+read through the `Dagster Docs <https://docs.dagster.io/getting-started>`__ to
+familiarize yourself with the tool's main concepts.
+
+There are a handful of Dagster concepts worth understanding prior
+to interacting with the PUDL data processing pipeline:
+
+**Dagit:**
+`Dagit <https://docs.dagster.io/concepts/dagit/dagit>`__ is the Dagster
+UI for monitoring and executing ETL runs.
+
+**Software Defined Assets (SDAs):**
+
+    *An asset is an object in persistent storage, such as a table, file, or
+    persisted machine learning model. A software-defined asset is a Dagster object that
+    couples an asset to the function and upstream assets that are used to produce
+    its contents.*
+
+`SDAs <https://docs.dagster.io/concepts/assets/software-defined-assets>`__
+are the computation building blocks in a Dagster project.
+Assets are linked together to form a direct acyclic graph (DAG) which can
+be execute to persist the data created by the assets. In PUDL, each asset
+is a dataframe written to SQLite or parquet files. Assets in PUDL can be
+raw extracted dataframes, partially cleaned tables or fully normalized
+tables.
+
+SDAs are created by applying the ``@asset`` decorator to a function.
+
+**IO Managers:**
+
+    *IO Managers are user-provided objects that store asset outputs
+    and load them as inputs to downstream assets.*
+
+Each asset has an `IO Manager  <https://docs.dagster.io/concepts/io-management/io-managers>`__
+that tells Dagster how to handle the objects returned by the software
+defined asset's underlying function. The IO Managers in PUDL read and
+write dataframes to and from databases and parquet files.
+
+**Resources:**
+
+   *Resources are objects that are shared across the implementations of
+   multiple software-defined assets and that can be plugged in
+   after defining those assets.*
```

Comment:
This doesn't make a lot of sense to me, but that might be because it's a difficult concept to explain. What is an implementation of an asset? I think the word implementation is tripping me up, not sure if it's a technical term here or not. Also confused by the phrase "plugged in"

Reply:
I agree this description is confusing. I removed it and tried to simplify the description.